### PR TITLE
Allow Addition of Cross without the  Mobject in argument

### LINKED ISSUE #1838: Allow `Cross` to be initialized without being based on an existing Mobject.
## Enhancement proposal

Currently Cross needs a mobject passed to it, I think it should be reworked to not necessarily require this. Also maybe easily specifying the length of the cross in the `__init__` too?

Current `Cross` code:
```py
class Cross(VGroup):
    def __init__(self, mobject, stroke_color=RED, stroke_width=6, **kwargs):
        VGroup.__init__(
            self,
            Line(UP + LEFT, DOWN + RIGHT),
            Line(UP + RIGHT, DOWN + LEFT),
        )
        self.replace(mobject, stretch=True)
        self.set_stroke(color=stroke_color, width=stroke_width)
```

## Additional comments

replacing `VGroup.__init__` with `super().__init__` too

### PR DESCRIPTION
**Pull request by asjadaugust**
## Motivation

The Cross class used to need Mobject instance to be initialized, however there was a need to create the Cross without a reference. 
Fixes #1838
## Explanation for Changes

Cross can be added to the Scene independent of the Mobject upon which the Cross need to be fit.
_The following example has 'stroke_color=BLUE' set_

Example3

Or if the Mobject is provided as arguments, it will fit to it.

Example1

If the scale_factor parameter is provided with or without the Mobject it will scale up to the specified value.

Example2
```python
from manim import *


class ExampleCross(Scene):
    def construct(self):
        text = Text("Manim")
        cross = Cross(text, scale_factor=2)
        self.add(cross, text)
```

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [x] The changelog entry is completed if necessary
- [x] Newly added functions/classes either have a docstring or are private
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/mobject/shape_matchers.py [hunk 1]
```diff
@@ -117,13 +119,44 @@ def get_fill_color(self):
...
 
 class Cross(VGroup):
-    def __init__(self, mobject, stroke_color=RED, stroke_width=6, **kwargs):
-        VGroup.__init__(
-            self,
-            Line(UP + LEFT, DOWN + RIGHT),
-            Line(UP + RIGHT, DOWN + LEFT),
+    """Creates a cross.
+
+     Parameters
```
Comment (kolibril13) [inline manim/mobject/shape_matchers.py:124]:
```suggestion
    Parameters
```

**Comment by kolibril13** [comment]
Thanks for the contribution!
Here is the docs page:
https://manimce--1847.org.readthedocs.build/en/1847/reference/manim.mobject.shape_matchers.Cross.html?highlight=cross#manim.mobject.shape_matchers.Cross
image
It looks a bit off, but I think this is only because of the wrong indentation of the parameters variable.
It should look similar to this one:
image

**Comment by kolibril13** [comment]
looks good!

**Review by kolibril13 — APPROVED** [review]

File: manim/mobject/shape_matchers.py [hunk 2]
```diff
@@ -117,13 +119,44 @@ def get_fill_color(self):
...
-            Line(UP + LEFT, DOWN + RIGHT),
-            Line(UP + RIGHT, DOWN + LEFT),
+    """Creates a cross.
+
+    Parameters
+    ----------
+    mobject
+        The mobject linked to this instance. It fits the mobject when specified. Defaults to None.
+    stroke_color
+        Specifies the color of the cross lines. Defaults to Red.
```
Comment (kolibril13) [inline manim/mobject/shape_matchers.py:129]:
```suggestion
        Specifies the color of the cross lines. Defaults to RED.
```

File: manim/mobject/shape_matchers.py [hunk 3]
```diff
@@ -117,13 +119,44 @@ def get_fill_color(self):
...
+                cross = Cross()
+                self.add(cross)
+    """
+
+    def __init__(
+        self,
+        mobject: Optional[Mobject] = None,
+        stroke_color: Optional[Color] = RED,
+        stroke_width: Optional[float] = 6,
+        scale_factor: Optional[float] = 1,
```
Comment (hydrobeam) [inline manim/mobject/shape_matchers.py:151]: `Optional` is for when `None` is also a valid paramter, in these cases, it's not. So Optional is not appropriate here.
```suggestion
        stroke_color: Color = RED,
        stroke_width: float = 6,
        scale_factor: float = 1,
```

File: manim/mobject/shape_matchers.py [hunk 4]
```diff
@@ -117,13 +119,44 @@ def get_fill_color(self):
...
+
+        class ExampleCross(Scene):
+            def construct(self):
+                cross = Cross()
+                self.add(cross)
+    """
+
+    def __init__(
+        self,
+        mobject: Optional[Mobject] = None,
```
Comment (hydrobeam) [inline manim/mobject/shape_matchers.py:148]: Doesn't explicitly need to be a `Mobject`, so quotes are required here.
```suggestion
        mobject: Optional["Mobject"] = None,
```

**Review by hydrobeam — COMMENTED** [review]
Typing comments

**Review by hydrobeam — APPROVED** [review]
This PR looks good to me! Thank you for handling the issue.

**Pull request merged**